daisy/step_includeworkflow.go

```diff
@@ -35,7 +35,7 @@ type IncludeWorkflow struct {
 func (i *IncludeWorkflow) populate(ctx context.Context, s *Step) DError {
 	if i.Path != "" {
 		var err error
-		if i.Workflow, err = s.w.NewIncludedWorkflowFromFile(i.Path); err != nil {
+		if i.Workflow, err = s.w.NewIncludedWorkflowFromFile(i.Path, i.Vars); err != nil {
```

Comment:
Is the loop over vars and adding them to the subworkflow below in this file needed, since it's already done by NewIncludedWorkflowFromFile (which calls readWorkflow which actually sets variables)?

Reply:
FromFile is not the only way. The sub/included workflow can actually be directly defined in json inline. Or, when we construct workflow in memory, similarly we can directly set a sub/included workflow.